diff --git a/train_ssl.py b/train_ssl.py
index e6d91a3..ea1adf2 100644
--- a/train_ssl.py
+++ b/train_ssl.py
@@ -50,7 +50,7 @@ from eval_ssl import evaluate, summary_evaluation_results
     help='Need if train is False. Weights need to match the model structure.'
 )
 @click.option(
-    '--batch_size', type=click.INT,required=True,default=4,
+    '--batch_size', type=click.INT,required=True,default=64,
     help='Batchsize.'
 )
 @click.option(
