diff --git a/configs/dino_config.yml b/configs/dino_config.yml
index 835d88d..bfbe10c 100644
--- a/configs/dino_config.yml
+++ b/configs/dino_config.yml
@@ -31,7 +31,7 @@ data_loader_parameter:
   shuffle: True  # only train
   drop_last: True  # only train
   batch_size: 256
-  num_workers: 12
+  num_workers: 4
 
 optimizer_parameter:
   lr: 0.0001
diff --git a/train_ssl.py b/train_ssl.py
index e6d91a3..ea1adf2 100644
--- a/train_ssl.py
+++ b/train_ssl.py
@@ -50,7 +50,7 @@ from eval_ssl import evaluate, summary_evaluation_results
     help='Need if train is False. Weights need to match the model structure.'
 )
 @click.option(
-    '--batch_size', type=click.INT,required=True,default=4,
+    '--batch_size', type=click.INT,required=True,default=64,
     help='Batchsize.'
 )
 @click.option(
